app/models/group_assignment.rb

```diff
@@ -10,6 +10,22 @@ class GroupAssignment < ApplicationRecord
   scope :ongoing, (-> { where('group_assignments.period_end > ?', Time.zone.today) })
   scope :no_end, (-> { where(period_end: nil) })
 
+
+  scope :start_within, ->(date_range) { start_before(date_range.last).start_after(date_range.first) }
+  scope :end_within, ->(date_range) { where(period_end: date_range) }
+  scope :end_after, ->(date) { where('period_end > ?', date) }
+  scope :end_before, ->(date) { where('period_end < ?', date) }
+
+  scope :start_after, ->(date) { where('period_start > ?', date) }
+  scope :start_before, ->(date) { where('period_start < ?', date) }
+
+  scope :active_between, lambda { |start_date, end_date|
+    no_end.start_before(end_date)
+          .or(
+            start_before(end_date).end_after(start_date)
+          )
+  }
```

Comment:
Maybe move these scopes into a new concern which could also be used in `Assignment`?